src/antsibull_core/subprocess_util.py

```diff
@@ -87,6 +100,22 @@ async def _stream_log(
     return "".join(lines)
 
 
+def _get_log_func_and_prefix(
+    name: str, loglevel: str | OutputCallbackType | None, logger: Any
+) -> tuple[OutputCallbackType | None, str]:
+    logfunc: Callable[[str], Any] | None = None
+    log_prefix = f"{name}: "
+    if loglevel:
+        if callable(loglevel):
+            logfunc = loglevel
+            log_prefix = ""
+        else:
+            logfunc = cast("Callable[[str], Any]", getattr(logger, loglevel))
+            if isinstance(logger, TwiggyLogger):
+                logfunc = _escape_brackets(logfunc)
```

Comment:
I wonder whether it's not better to do something like
```suggestion
                logfunc = lambda msg: logfunc("{0}", msg)
```
instead.

Also for the standard Python logger you also have to watch out, and do something like `logfunc = lambda msg: logfunc("%s", msg)`. Otherwise `%` formatting in the log message will be interpreted.